x/cork/keeper/keeper.go

```diff
@@ -222,6 +303,56 @@ func (k Keeper) GetApprovedCorks(ctx sdk.Context, threshold sdk.Dec) (approvedCo
 	return winningCorks
 }
 
+func (k Keeper) GetApprovedScheduledCorks(ctx sdk.Context, currentBlockHeight uint64, threshold sdk.Dec) (approvedCorks []types.Cork) {
+
+	corksForBlockHeight := make(map[uint64][]types.Cork)
+	var corkPowersForBlockHeight map[uint64][]int64
+
+	totalPower := k.stakingKeeper.GetLastTotalPower(ctx)
+
+	k.IterateScheduledCorks(ctx, func(val sdk.ValAddress, scheduledBlockHeight uint64, addr common.Address, cork types.Cork) (stop bool) {
```

Comment:
Similar comment here, I wonder if we couldn't use a normalization strategy to make this an O(1) lookup, i.e. storing something in the keeper indexed by a vote end boundary (say block 10), and given any input block height you can store it by normalizing it up to a vote boundary when storing it and normalizing the current block height down when doing the lookup. The side effect here would be that all scheduled corks wait for the beginning of the next vote period rather than using their exact scheduled height, but you would be able to avoid iterating.